Sources/Profile-Modify.php

```diff
@@ -2892,7 +2892,7 @@ function profileValidateSignature(&$value)
 
 		$unparsed_signature = strtr(un_htmlspecialchars($value), array("\r" => '', '&#039' => '\''));
 		// Too long?
-		if (!empty($sig_limits[1]) && $smcFunc['strlen']($unparsed_signature) > $sig_limits[1])
+		if (!empty($sig_limits[1]) && (!empty($sig_bbc) ? ($smcFunc['strlen']($unparsed_signature) - $smcFunc['strlen']($sig_bbc)) : $smcFunc['strlen']($unparsed_signature)) > $sig_limits[1])
```

Comment:
Already have one in the JavaScript that I added. If you look in the Profile.template.php you'll see a parseBaby method, that strips "All" bbcode that has a matching end tag. So if you have "[k]Hello[/b]" it would count all of that as you don't have the correct starting tag. But if you had "[b]Hello[/b]" it would remove the bold tags and only count the 5 letters for hello.